python/context.cpp

```diff
@@ -48,13 +50,16 @@ struct proc_allocation_shim {
         gpu_id = py2optional<int>(gpu, "gpu_id must be None, or a non-negative integer", is_nonneg());
     };
 
-    void set_num_threads(int threads) {
-        pyarb::assert_throw([](int n) { return n>0; }(threads), "threads must be a positive integer");
-        num_threads = threads;
+    void set_num_threads(unsigned threads) {
```

Comment:
This function needs to be documented.

Reply:
You mean `set_num_threads()`? You mean, a note in the docs `arbor.proc_allocation` that the minimum number is 1?

Reply:
I mean that the class is not fully documented: https://docs.arbor-sim.org/en/latest/python/hardware.html#arbor.proc_allocation 
I'm now noticing it's not just `set_num_threads`, but also `get_num_threads` and `get_gpu_id`. 

Also, you've made changes to `context`, so the documentation for that too needs updating: https://docs.arbor-sim.org/en/latest/python/hardware.html#arbor.context

Reply:
The Python class documentation is on lines 106-123, not on the shim. The Python binding combines `set_num_threads` and `get_num_threads` into `threads`, I've update that and `doc/python/hardware.rst` to reflect 1 is the minimum. Context docs is also corrected (whoops!)